src/compiler.rs

```diff
@@ -94,7 +93,9 @@ pub fn compile_sync(
   send_future.wait().unwrap();
 
   let recv_future = resources::worker_recv_message(compiler.rid);
-  let res_msg = recv_future.wait().unwrap().unwrap();
+  let result = recv_future.wait().unwrap();
```

Comment:
The next error I run into in tests is pretty complicated. For some reason we end up with a thread panic in the middle of our file not found error print out:
```
Compiling file:///home/andy/Documents/projects/personal/deno/tests/error_004_missing_module.ts
thread 'Uncaught NotFound: Cannot resolve module "bad-module.ts" from "/home/andy/Documents/projects/personal/deno/tests/error_004_missing_module.ts"
    at DenoError (deno/js/errors.ts:22:5)
    at maybeError (deno/js/errors.ts:33:12)
    at maybeThrowError (deno/js/errors.ts:39:15)
    at sendSync (deno/js/dispatch.ts:87:5)
    at fetchModuleMetaData (deno/js/os.ts:81:19)
    at _resolveModule (deno/js/compiler.ts:249:38)
    at resolveModuleNames (deno/js/compiler.ts:479:35)
    at compilerHost.resolveModuleNames (deno/third_party/node_modules/typescript/lib/typescript.js:118650:138)
    at resolveModuleNamesWorker (deno/third_party/node_modules/typescript/lib/typescript.js:86767:127)
    at resolveModuleNamesReusingOldState (deno/third_party/node_modules/typescript/lib/typescript.js:87001:24)tokio-runtime-worker-0
' panicked at 'called `Result::unwrap()` on an `Err` value: DenoError { repr: Simple(BadResource, "bad resource id") }', src/libcore/result.rs:997:5
```
It's not really a problem for end users, but it just makes it very difficult to test expected failures. The best way to replicate this one is with `target/release/deno tests/error_004_missing_module.ts --reload`.
It looks like the thread that `compile_sync` is running in manages to panic here before we finish our error print and exit in `workers.rs`. Maybe we should avoid the panic here by returning a Result error.